rllib/models/action_dist.py

```diff
@@ -99,3 +104,75 @@ def required_model_output_shape(
                 required input vector (minus leading batch dimension).
         """
         raise NotImplementedError
+
+
+@DeveloperAPI
+class MultiActionDistributionMixIn(ActionDistribution):
```

Comment:
one thing I am curious is why does this have to be a MixIn, instead of having all this logic as part of MultiActionDistribution class?

Reply:
> one thing I am curious is why does this have to be a MixIn, instead of having all this logic as part of MultiActionDistribution class?

The point here is to reduce duplicate code. All these methods in `MultiActionDistributionMixIn` are shared for `MultiActionDistribution` for TensorFlow, PyTorch, and maybe Jax in the future.